drivers/display/ls0xx.c

```diff
@@ -0,0 +1,496 @@
+/*
+ * Copyright (c) 2020 Rohit Gujarathi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT   sharp_ls0xx
+
+#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ls0xx);
+
+#include <string.h>
+#include <device.h>
+#include <drivers/display.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+/* Supports LS012B7DD01, LS012B7DD06, LS013B7DH03, LS013B7DH05
+ * LS013B7DH06, LS027B7DH01A, LS032B7DD02, LS044Q7DH01
+ */
+
+/* Note:
+ * -> .write API should be called atleast once per second so that VCOM can
+ *      be toggled. If there is nothing to update then buf can be set to
+ *      NULL or, width/height can be set to 0 so the driver will only
+ *      toggle VCOM and return
+ * -> high/1 means white, low/0 means black
+ * -> Display expects LSB first
+ * -> If devicetree contains DISP pin then it will be set to high
+ *      during initialization.
+ * -> Sharp memory displays require toggling the VCOM signal periodically
+ *      to prevent a DC bias ocurring in the panel. The DC bias can damage
+ *      the LCD and reduce the life. This signal must be supplied
+ *      externally from either serial input or an external signal on
+ *      the EXTCOMIN pin.
```

Comment:
An other use case for #23075?

Reply:
@vanwinkeljan , using a k_timer is becoming very complicated. To avoid contention on spi bus i am trying to use a mutex. but since the expiry_fn of k_timer is called in an isr i am not able to use a mutex. So i might have to use a workqueue. 

Alternatively i suggest using a K_THREAD. What do you think?

Reply:
> Alternatively i suggest using a K_THREAD. What do you think?

Sounds like a reasonable alternative


@jfischer-phytec-iot 
 > An other use case for #23075?

Combined with the queue manager mentioned in that issues this driver sounds like a perfect candidate.

Reply:
Hello @vanwinkeljan ,

This k_thread/k_timer (any asynchronous method really, asynchronous to the write command) approach has one flaw, which may or may not need to be addressed.
If the user is using the same SPI interface for this display and some other sensor. Then there might be contention during a sensor read/write and VCOM toggle command over SPI.

To resolve this, we can either 
1) limit the user to not using the same SPI interface by adding a note in the shields documentation.
2) Use the current approach of using the generic write API if the user selects LS0XX_EXTMODE_SERIAL=y and use a thread for GPIO toggling if LS0XX_EXTMODE_SERIAL=n.

Reply:
@rgujju Options 2 sounds the best solution until we have something like the async sequence manager

Reply:
yes! I have gone ahead with option 2 in the latest commits.